superset-frontend/src/explore/components/ControlPanelSection.tsx

```diff
@@ -0,0 +1,206 @@
+  'controlSetRows'
+> & {
+  controlSetRows: ExpandedControlItem[][];
+};
+
+export const iconStyles = css`
+  &.anticon {
+    font-size: unset;
+    .anticon {
+      line-height: unset;
+      vertical-align: unset;
+    }
+  }
+`;
+
+type IControlPanelSectionProps = {
+  actions: Partial<ExploreActions> & Pick<ExploreActions, 'setControlValue'>;
+  sectionId: string;
+  section: ExpandedControlPanelSectionConfig;
+  hasErrors: boolean;
+  errorColor: string;
+  renderControl: (item: CustomControlItem) => JSX.Element;
+  isDisabled?: boolean;
+};
+
+export function ControlPanelSection({
+  actions: { setControlValue },
+  sectionId,
+  section,
+  hasErrors,
+  errorColor,
+  renderControl,
+  isDisabled,
+  ...restProps // https://github.com/react-component/collapse/issues/73#issuecomment-323626120
+}: IControlPanelSectionProps) {
+  const { label, description, disabledTooltipText } = section;
+  const wasDisabled = usePrevious(isDisabled);
+  const clearDisabledSectionControls = useCallback(() => {
+    ensureIsArray(section.controlSetRows).forEach(controlSets => {
+      controlSets.forEach(controlItem => {
+        if (
+          controlItem &&
+          !React.isValidElement(controlItem) &&
+          controlItem.name &&
+          controlItem.config &&
+          controlItem.name !== 'datasource'
+        ) {
+          setControlValue?.(controlItem.name, controlItem.config.default);
+        }
+      });
+    });
+  }, [setControlValue, section]);
+
+  useEffect(() => {
+    if (wasDisabled === false && isDisabled) {
+      clearDisabledSectionControls();
+    }
+  }, [wasDisabled, isDisabled]);
+
+  const PanelHeader = () => (
+    <span data-test="collapsible-control-panel-header">
+      <span
+        css={(theme: SupersetTheme) => css`
+          font-size: ${theme.typography.sizes.m}px;
+          line-height: 1.3;
+        `}
+      >
+        {label}
+      </span>{' '}
+      {!isDisabled && description && (
+        <Tooltip id={sectionId} title={description}>
+          <Icons.InfoCircleOutlined css={iconStyles} />
+        </Tooltip>
+      )}
+      {!isDisabled && hasErrors && (
+        <Tooltip
+          id={`${kebabCase('validation-errors')}-tooltip`}
+          title={t('This section contains validation errors')}
+        >
+          <Icons.InfoCircleOutlined
+            css={css`
+              ${iconStyles};
+              color: ${errorColor};
+            `}
+          />
+        </Tooltip>
+      )}
+      {isDisabled && disabledTooltipText && (
+        <Tooltip
+          id={`${kebabCase('disabled-section')}-tooltip`}
+          title={disabledTooltipText}
+        >
+          <Icons.InfoCircleOutlined
+            data-test="disabled-section-tooltip"
+            css={css`
+              ${iconStyles};
+            `}
+          />
+        </Tooltip>
+      )}
+    </span>
+  );
+
+  return (
+    <Collapse.Panel
+      {...restProps}
+      collapsible={isDisabled ? 'disabled' : undefined}
```

Comment:
Not sure why but for me the disabled panel still slightly moves when you click on it:

https://user-images.githubusercontent.com/13007381/207926655-ad23e15d-c1aa-44b0-9117-6b79af7c9694.mov

Reply:
Wow! Nice catch!